var/spack/repos/builtin/packages/py-smartredis/package.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PySmartredis(PythonPackage):
+    """A Python Interface for the SmartRedis Library Client"""
+
+    homepage = "https://www.craylabs.org/docs/smartredis.html"
+    pypi = "smartredis/smartredis-0.4.0.tar.gz"
+    git = "https://github.com/CrayLabs/SmartRedis"
+
+    maintainers("MattToast")
+
+    version("0.4.1", sha256="fff16ed1eb09648ac3c3f845373beb37f3ffe7414d8745ae36af9daf585f8c5b")
+    version("0.4.0", sha256="d12779aa8bb038e837c25eac41b178aab9e16b729d50ee360b5af8f813d9f1dd")
+
+    depends_on("python@3.8:3.10", type=("build", "run"))
```

Comment:
According to [setup.cfg#L37](https://github.com/CrayLabs/SmartRedis/blob/v0.4.1/setup.cfg#L37) python 3.7 is also supported. Is there any reason why you restricted it to 3.8`

Reply:
I wasn't sure if I should still include python 3.7 in a new package since it has reached EOL, even if it is supported upstream. More than willing to add it in though!

Reply:
You can add it, 3.7 is deprecated but won't be removed entirely until the next minor release (approximately November).